### ISSUE TITLE
Grid Editor - Problem Saving Core Map

### ISSUE THREAD

**Issue by john-science** [terrapower/armi#490]
@jakehader Reported this bug: given the right conditions (which are common), the grid editor is failing to write output files.

## Steps to Reproduce the Bug

1. From the current "master" branch, open the grid editor on the command line: `$ armi grids`
2. Load this blueprints YAML file: [test-full-bp.zip](https://github.com/terrapower/armi/files/7598401/test-full-bp.zip)
3. Hit the buttons "Expand to full core" >> "Apply" >> "Save to grid blueprints..."
4. This will yield a valid output YAML file that looks like:

```yaml
core:
  geom: hex
  lattice map:
  grid bounds:
  symmetry: full
  grid contents:
    ? - 0
      - 0
    : IF
```
5. Now, add full ring of 6 "inner fuel"s around the central hex. and again "Save to grid blueprints..." You will now see the broken output YAML file:

```yaml
core:
  geom: hex
  lattice map: |-
    None
  grid bounds:
  symmetry: full
  grid contents:
```

## The Bug

The second core map above is wrong because it has no output data.

## What Caused this Bug?

Jake and determined this bug entered the "master" branch at this PR: https://github.com/terrapower/armi/pull/439

**Comment by john-science** [comment]
@youngmit  Before I spend too much more time on this, do you happen to understand the issue?  I know the bug entered the codebase during the above PR, and that PR _did_ hit [gridEditor.py](https://github.com/terrapower/armi/pull/439/files#diff-f5b03b844708a59fd3cbeb46d1667a49ca885ba8bbfb3e22a4de20c6554d1f8d).

If nothing jumps out at your immediately, I'll start debuggin.

**Comment by jakehader** [comment]
PR #440 might also be influencing this behavior (@ntouran) with incorrectly writing the full core hex geometry.

**Comment by ntouran** [comment]
If you determined that the bug entered at #439, don't we know it's not #440?

Also, are there any error messages or anything?

**Comment by john-science** [comment]
@jakehader  @ntouran   I am fairly sure the bug entered the master branch at PR #439

**Comment by jakehader** [comment]
This is where the error that occurs from catching a generic exception - https://github.com/terrapower/armi/blob/b94744fdf12f247828dd07bf99e251651b494a13/armi/reactor/blueprints/gridBlueprint.py#L592

I believe it has to do with some rows in the dictionary having only separators `'-'` and then it fails to write the lattice map. This used to then default to writing the grid contents if the lattice map failed to write but something about the change in #439 changed this behavior.

**Comment by jakehader** [comment]
Fix in #494

**Comment by ntouran** [comment]
Ok so we still do get an exception when we run the reproduction example but with the exception no longer being hidden:

```
The `lattice map` for the current assembly arrangement cannot be written. Defaulting to using the `grid contents` dictionary instead. Exception: max() arg is an empty sequence
```

This is very likely coming from `asciimaps._updateDimensionsFromData`. That may help us find the root cause.
